django/db/backends/sqlite3/base.py

```diff
@@ -554,19 +554,29 @@ def _sqlite_format_dtdelta(conn, lhs, rhs):
     LHS and RHS can be either:
     - An integer number of microseconds
     - A string representing a datetime
+    - A scalar factor (e.g. float)
     """
-    try:
-        real_lhs = datetime.timedelta(0, 0, lhs) if isinstance(lhs, int) else backend_utils.typecast_timestamp(lhs)
-        real_rhs = datetime.timedelta(0, 0, rhs) if isinstance(rhs, int) else backend_utils.typecast_timestamp(rhs)
-        if conn.strip() == '+':
-            out = real_lhs + real_rhs
+    if conn.strip() in ['+', '-']:
+        try:
+            real_lhs = datetime.timedelta(0, 0, lhs) if isinstance(lhs, int) else backend_utils.typecast_timestamp(lhs)
```

Comment:
I would add a separate hook for this (in a separate commit), e.g.

```python

def _sqlite_prepare_dtdelta_param(conn, param):
    if conn in ['+', '-']:
        if isinstance(param, int):
            return datetime.timedelta(0, 0, param)
        else:
            return backend_utils.typecast_timestamp(param)
    if conn in ['*', '/'] and not isinstance(param, (int, float)): 
        raise TypeError
    return param    

@none_guard
def _sqlite_format_dtdelta(conn, lhs, rhs):
    """
    LHS and RHS can be either:
    - An integer number of microseconds
    - A string representing a datetime
    - A scalar factor (e.g. float)
    """
    conn = conn.strip()
    try: 
        real_lhs = _sqlite_prepare_dtdelta_param(conn, lhs)
        real_rhs = _sqlite_prepare_dtdelta_param(conn, rhs)
        if conn == '+':
            # typecast_timestamp returns a date or a datetime without timezone.
            # It will be formatted as "%Y-%m-%d" or "%Y-%m-%d %H:%M:%S[.%f]"
            out = str(real_lhs + real_rhs)
        elif conn == '-':
            out = str(real_lhs - real_rhs)
        elif conn == '*':
            out = int(real_lhs * real_rhs)
        else:
            out = int(real_lhs / real_rhs)
    except (ValueError, TypeError):
        return None
    else:
        return out
```